content/courses/onchain-development/local-setup.md

```diff
@@ -2,116 +2,219 @@
+can be any number).
+
+### Running the Solana Test Validator
 
-#### Download Anchor
+The Solana Test Validator is a local emulator for the Solana blockchain. It
+provides developers with a private and controlled environment to build and test
+Solana programs without needing to connect to a public testnet or mainnet.
 
-Finally [download Anchor](https://www.anchor-lang.com/docs/installation):
+To start the Solana Test Validator, run the following command:
 
+```bash
+solana-test-validator
 ```
+
+For more detailed information, you can refer to the
+[Solana Test Validator guide](https://solana.com/developers/guides/getstarted/solana-test-validator).
+
+### Download Anchor
+
+Finally, [download Anchor](https://www.anchor-lang.com/docs/installation):
+
+```bash
 cargo install --git https://github.com/coral-xyz/anchor avm --locked --force
 avm install latest
 avm use latest
 ```
 
-Afterwards, `anchor -V` should show `anchor-cli 0.30.0`.
+After installation, `anchor -V` should display `anchor-cli 0.30.1`. For more
+detailed information on Anchor, refer to
+[The Anchor Book](https://book.anchor-lang.com).
 
-#### Check your Anchor installation
+### Verify your Anchor Installation
 
-Create a temporary project, with the default contents, using Anchor and make
-sure it compiles and runs our tests:
+Create a temporary project with the default contents using Anchor and ensure it
+compiles and runs:
 
 ```bash
 anchor init temp-project
 cd temp-project
 anchor test
 ```
 
-**The `anchor test` command should complete with no errors or warnings**.
-However you may encounter issues, and we'll fix them below:
+If you are running `solana-test-validator`, you may encounter the error
+`Error: Your configured rpc port: 8899 is already in use` when running
+`anchor test`. To resolve this, stop the `solana-test-validator` before running
+`anchor test`.
 
-##### `package `solana-program v1.18.12` cannot be built because it requires rustc 1.75.0 or newer` error
+**The `anchor test` command should complete without errors or warnings**. If you
+encounter issues, we’ll address them below:
+
+#### Error: `package 'solana-program v1.18.12' cannot be built because it requires rustc 1.75.0 or newer`
 
 Run `cargo add solana-program@"=1.18.x"`, where `x` matches your version of
 `solana-cli`. Then re-run `anchor test`.
 
-##### `Error: Unable to read keypair file`
+#### Error: `Unable to read keypair file`
 
 Add a keypair to `.config/solana/id.json`. You can either copy a keypair from an
 `.env` file (just the array of numbers) into a file or use the command
 `solana-keygen new --no-bip39-passphrase` to create a new keypair file. Then
 re-run `anchor test`.
 
-##### `unused variable: 'ctx'` warning
+#### Error: `Function _ZN112_$LT$solana_program..instruction..InstructionError$u20$as$u20$solana_frozen_abi..abi_example..AbiEnumVisitor$GT$13visit_for_abi17ha3a75fe06f2a2af4E Stack offset of 4584 exceeded max offset of 4096 by 488 bytes, please minimize large stack variables`
+
```

Comment:
This would be a good addition to an upgrade guide for existing programs, but not to a system setup guide - Cargo.lock should exist by default, and have current dependencies. I don't think this error would apply for programs just made with `anchor init`.

Reply:
I recently faced this issue while setting up an anchor project cloned from github with the latest anchor version. But understood this will not be applicable for the local setup guide. I have deleted this. If we have any other program upgrade guide will add it there.